topk-js/src/data/function_expr.rs

```diff
@@ -0,0 +1,118 @@
+use napi::bindgen_prelude::*;
+use napi_derive::napi;
+
+use super::value::{BinaryVector, Vector};
+
+#[napi]
+#[derive(Debug, Clone)]
+pub enum FunctionExpression {
+    KeywordScore,
+    VectorScore { field: String, query: VectorQuery },
+    SemanticSimilarity { field: String, query: String },
+}
+
+#[napi(namespace = "query")]
+pub fn semantic_similarity(field: String, query: String) -> FunctionExpression {
+    FunctionExpression::SemanticSimilarity { field, query }
+}
+
+#[napi]
+#[derive(Debug, Clone)]
+pub enum VectorQuery {
+    F32 { vector: Vec<f64> },
+    U8 { vector: Vec<u8> },
+}
+
+impl Into<topk_protos::v1::data::Vector> for VectorQuery {
+    fn into(self) -> topk_protos::v1::data::Vector {
+        match self {
+            VectorQuery::F32 { vector } => {
+                topk_protos::v1::data::Vector::float(vector.iter().map(|v| *v as f32).collect())
+            }
+            VectorQuery::U8 { vector } => topk_protos::v1::data::Vector::byte(vector),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum VectorDistanceQuery {
+    Vector { vector: Vector },
+    Binary { vector: BinaryVector },
```

Comment:
why is `Binary` a separate option here instead of doing `Vector { vector: Vector::Binary {} }` just like for the other cases?